src/chain/public-key.ts

```diff
@@ -68,6 +68,14 @@ export class PublicKey implements ABISerializableObject {
         this.data = data
     }
 
+    /**
+     * Returns the core 33-byte compressed public key data as a Uint8Array.
+     * This is suitable for cryptographic operations like verification.
+     */
+    getCompressedKeyBytes(): Uint8Array {
```

Comment:
is it worth throwing an error when `this.data.array` is less than 33 bytes? Either we do that or we add a check in the constructor that throws an error when `this.data` is not the correct length.

Reply:
I am not sure whether we should improve the error handling in this PR. It would change the designed behavior of the library. Illegal bytes passed in situation was not considered at all in this library.

Reply:
Yeah, maybe it should be something that this library checks, but you're right that doesn't need to be part of this PR 👍